core/domain/suggestion_services.py

```diff
@@ -412,6 +413,39 @@ def accept_suggestion(
         _update_user_proficiency(user_proficiency)
 
 
+def update_suggestion(suggestion_id, translation_html):
+    """Updates a suggestion with the given suggestion_id and translation_html.
+
+    Args:
+        suggestion_id: str. The id of the suggestion to be accepted.
+        translation_html: str. The new translation html.
+
+    Raises:
+        Exception. The suggestion is already handled.
+        Exception. The suggestion is not valid.
+
+    Returns:
+        _update_suggestion(suggestion). Updates the suggestion.
+    """
+
+    suggestion = get_suggestion_by_id(suggestion_id)
+
+    if suggestion is None:
+        raise utils.ValidationError(
+            'You cannot change the suggestion with id %s because it does '
+            'not exist.' % (suggestion_id)
+        )
+    if suggestion.is_handled:
+        raise utils.ValidationError(
+            'The suggestion with id %s has already been accepted/'
+            'rejected.' % (suggestion_id)
+        )
+
+    suggestion.change.translation_html = translation_html
```

Comment:
Make sure the html is cleaned via html_cleaner* (I'm not sure whether it's done in the suggestion.validation method)

Reply:
I used the same appeoach which were used to update `change` object with html content. Even `ResubmitSuggestionHandler` uses the same approach to update `change` objects. I searched in the code base where html_cleaner is being used and found it was used for format images which were uploaded. Since updating translation suggestions do not allow users to add images, is it necessary to clean via html_cleaner? Grateful if you could help me on this as well.

Reply:
Sorry for the misunderstanding, I wanted to say html_cleanerhtml_cleaner.clean function*

@sagangwee We should use it in general for the suggestions having HTML data (maybe in the validation method), should we create an issue and fix it separately?

Reply:
@DubeySandeep Thanks for raising this! Created an issue here: https://github.com/oppia/oppia/issues/12482.